openedx/core/lib/cache_utils.py

```diff
@@ -1,12 +1,50 @@
 """
 Utilities related to caching.
 """
+import collections
 import cPickle as pickle
 import functools
 import zlib
 from xblock.core import XBlock
 
 
+class Memoized(object):
+    """
+    Decorator. Caches a function's return value each time it is called.
+    If called later with the same arguments, the cached value is returned
+    (not reevaluated).
+
+    https://wiki.python.org/moin/PythonDecoratorLibrary#Memoize
+    """
+    def __init__(self, func):
+        self.func = func
+        self.cache = {}
```

Comment:
(Because it's never going to be emptied, as far as I can tell. Using the RequestCache as your cache would prevent that leaking, but would be reset after every request. If you need longer lived caching, we should think about maybe using the RequestCache-over-Django cache that @tobz and @ormsbee were working on)